pkg/apis/workspaces/v1alpha2/volumeComponent.go

```diff
@@ -11,4 +11,44 @@ type Volume struct {
 	// +optional
 	// Size of the volume
 	Size string `json:"size,omitempty"`
+	// Persistent defines whether to use persistent storage for this volume. Defaults
+	// to true. When set to false, storage is tied to a pod lifecycle and is erased on
+	// delete.
+	// +optional
+	// +kubebuilder:default=true
+	Persistent bool `json:"persistent,omitempty"`
```

Comment:
Can it be used together with `External`? Should we deny it on the implementation level?

Reply:
Yeah I agree. One thing we discussed is that semantically it only makes sense to have either [persistent, size] or [external] but not both.

I also need to remove the readonly parameter for now, it doesn't make sense in its current spot.

Reply:
We can revisit this topic once #324 is merged and if we decide to go ahead on #325.